src/software/SfM/main_SfMInit_ImageListing.cpp

```diff
@@ -407,38 +436,99 @@ int main(int argc, char **argv)
       focalPix = -1.0;
     }
 
-
     // Build intrinsic parameter related to the view
     std::shared_ptr<IntrinsicBase> intrinsic (NULL);
     if (focalPix > 0 && ppx > 0 && ppy > 0 && width > 0 && height > 0)
     {
+      EINTRINSIC camera_model = e_User_camera_model;
+      // If no user input choose a default camera model
+      if(camera_model == PINHOLE_CAMERA_START)
+      {
+        // Use standard lens with radial distortion by default
+        camera_model = PINHOLE_CAMERA_RADIAL3;
+        if(resizedImage)
+        {
+          // If the image has been resized, we assume that it has been undistorted
+          // and we use a camera without lens distortion.
+          camera_model = PINHOLE_CAMERA;
+        }
+        else if(focalLength_mm > 0.0 && focalLength_mm < 15)
+        {
+          // If the focal lens is short, the fisheye model should fit better.
+          camera_model = PINHOLE_CAMERA_FISHEYE;
+        }
+      }
+
       // Create the desired camera type
-      switch(e_User_camera_model)
+      switch(camera_model)
```

Comment:
maybe we can refactor to use the `createPinholeIntrinsic()` function that has been added. We only need to check for goPro that way